drivers/regulator/regulator_pmic.c

```diff
@@ -273,15 +273,13 @@ static int disable_regulator(const struct device *dev)
 	if (rc == 0) {
 		rc = -EINVAL;
 		return onoff_sync_finalize(&data->srv, key, NULL, rc, false);
+	} else if (rc == 1) {
+		/* Disable regulator */
+		dis_val = config->enable_inverted ? config->enable_val : 0;
+		rc = regulator_modify_register(config, config->enable_reg,
+				config->enable_mask, dis_val);
 	}
-	dis_val = config->enable_inverted ? config->enable_val : 0;
-	rc = regulator_modify_register(config, config->enable_reg,
-			config->enable_mask, dis_val);
-	if (rc) {
-		/* Error writing configs */
-		return onoff_sync_finalize(&data->srv, key, NULL, rc, true);
-	}
-	return onoff_sync_finalize(&data->srv, key, NULL, rc, true);
+	return onoff_sync_finalize(&data->srv, key, NULL, rc, false);
```

Comment:
This is a bug fix and really ought to go into a separate PR.

Reply:
Done, PR number is #51092